src/Sylius/Bundle/CurrencyBundle/Doctrine/ORM/ExchangeRateRepository.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CurrencyBundle\Doctrine\ORM;
+
+use Doctrine\ORM\NonUniqueResultException;
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+use Sylius\Component\Currency\Model\CurrencyInterface;
+use Sylius\Component\Currency\Model\ExchangeRateInterface;
+use Sylius\Component\Currency\Repository\ExchangeRateRepositoryInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class ExchangeRateRepository extends EntityRepository implements ExchangeRateRepositoryInterface
+{
+    /**
+     * @param CurrencyInterface $firstCurrency
+     * @param CurrencyInterface $secondCurrency
+     *
+     * @return ExchangeRateInterface|null
+     *
+     * @throws NonUniqueResultException
+     */
+    public function findOneWithCurrencyPair(CurrencyInterface $firstCurrency, CurrencyInterface $secondCurrency)
+    {
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        return $queryBuilder
+            ->where(
+                $queryBuilder->expr()->andX(
+                    $queryBuilder->expr()->eq('o.sourceCurrency', $firstCurrency->getId()),
+                    $queryBuilder->expr()->eq('o.targetCurrency', $secondCurrency->getId())
```

Comment:
Pretty sure i already tried that, but had to check again.

Turns out, while it would work with `setParameter` as it has a whole infrastructure to resolve what type was passed; with `eq`, the parameter simply gets cast to string, resulting in:

![image](https://cloud.githubusercontent.com/assets/9448101/20544526/c52f7ad8-b10a-11e6-8eb6-ac25ec1b99c8.png)